.github/workflows/build-asset.yml

```diff
@@ -0,0 +1,107 @@
+          node-version: '${{ env.NODEJS_VERSION }}'
+      - name: Record Node version
+        run: |
+          echo "NODEJS_VERSION=$(node -p process.versions.node)" >> $env:GITHUB_ENV
+          echo "NODEJS_VERSION=$(node -p process.versions.node)" >> "${GITHUB_ENV}"
+      - run: npm ci
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - if: ${{ matrix.os == 'windows-latest' }}
+        run: choco install nasm
+      - if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: sudo apt-get update
+      - if: ${{ matrix.os != 'windows-latest' }}
+        uses: hendrikmuhs/ccache-action@v1.2
+        with:
+          create-symlink: true
+          key: ${{ github.job }}-${{ matrix.os }}-${{ matrix.version }}
+      - if: ${{ matrix.os == 'windows-latest' }}
+        run: |
+          echo "NEXE_ASSET=${{ env.NEXE_ASSET }}.exe" >> $env:GITHUB_ENV
+          echo "EXECUTABLE_SUFFIX=.exe" >> $env:GITHUB_ENV
+      - name: Build and test asset
+        run: node tasks/asset-build && node tasks/asset-test-build && npm run test:integration:run
+      - name: Update release artefact name
+        shell: bash
+        run: |
+          artefact_name="nexe-asset-${{ matrix.os }}-${NODEJS_VERSION}${EXECUTABLE_SUFFIX}"
+          mv "${NEXE_ASSET}" "${artefact_name}"
+          echo "NEXE_ASSET=${artefact_name}" >> "${GITHUB_ENV}"
+          echo "NEXE_ASSET=${artefact_name}" >> $env:GITHUB_ENV
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.NEXE_ASSET }}
+          path: ${{ env.NEXE_ASSET }}
+          if-no-files-found: error
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: integration-tests-${{ matrix.os }}-${{ env.NODEJS_VERSION }}${{ env.EXECUTABLE_SUFFIX }}
+          path: integration-tests${{ env.EXECUTABLE_SUFFIX }}
+          if-no-files-found: error
+      - if: ${{ matrix.os == 'ubuntu-latest' }}
+        name: musl static build
+        run: |
+          set -x
+          export MUSL_BUILD=yes NEXE_TMP=${{ env.NEXE_TMP_CACHE_PATH }}/musl
+          eval $(npx -p node-musl musl-exports)
+          export CC="ccache ${CC}" CXX="ccache ${CXX}" LD="ccache ${LD}"
+          node tasks/asset-build
+          node tasks/asset-test-build
+          npm run test:integration:run
+      - if: ${{ matrix.os == 'ubuntu-latest' }}
+        name: Upload Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.NEXE_ASSET }}
+          path: ${{ env.NEXE_ASSET }}
+          if-no-files-found: error
+      - if: ${{ matrix.os == 'ubuntu-latest' }}
+        name: Upload Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: integration-tests-musl-${{ matrix.os }}-${{ env.NODEJS_VERSION }}${{ env.EXECUTABLE_SUFFIX }}
+          path: integration-tests
+          if-no-files-found: error
+      - if: startsWith(github.ref, 'refs/tags/')
```

Comment:
This was just a guess at a plan for how to handle this :-) Similar for the defaulting to prerelease below too :-)